ConsoleApplication2/ConsoleApplication2/ConsoleApplication2.cpp

```diff
@@ -0,0 +1,259 @@
+// ConsoleAplenplication2.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+#include <iostream>
+#include <exception>
+#include <algorithm>
+
+#include <functional>
+#include <array>
+#include <string_view>
+
+
+#define ALLOCATE_BYTES    10
+
+typedef unsigned int index_t;
+
+class SString {
+public:
+    SString() {
+        this->str = nullptr;
+        this->len = 0;
+        this->cap = 0;
+    }
+
+    SString(const char* in) : SString() {
+        while (in[this->len] != '\0') {
+            this->len++;
+        }
+        if (this->len > 0) {
+            reallocStorage(this->len + 1);
+
+            copyString(this->str, in, this->len + 1);
+        }
+    }
+
+    SString(const SString& in) : SString() {
+        this->len = in.len;
+        if (this->len > 0) {
+            reallocStorage(this->len + 1);
+
+            copyString(this->str, in.str, this->len + 1);
+        }
+    }
+
+    SString(SString&& in) : SString() {
+        this->len = in.len;
+        if (this->len > 0) {
+            reallocStorage(this->len + 1);
+
+            copyString(this->str, in.str, this->len + 1);
+        }
+
+        in.str = nullptr;
+    }
+
+    ~SString() {
+        std::cout << "memory free" << std::endl;
+        delete[] this->str;
+    }
+
+    // return curent symbol from string
+    char& operator[] (index_t ind) {
+        return this->str[ind];
+    }
+
+    // return current symbol from string with access of range
+    char& at(index_t ind);
+
+    void append(const char* in);
+
+    void append(const char* in, size_t num);
+
+    void append(const char* in, index_t start, index_t finish);
+
+    size_t size() {
+        return this->len;
+    }
+
+    size_t capacity() {
+        return this->cap;
+    }
+
+    SString& operator= (const SString& in) {
+        this->len = in.len;
+        this->cap = 0;
+
+        reallocStorage(this->len + 1);
+
+        copyString(this->str, in.str, this->len + 1);
+
+        return *this;
+    }
+
+    SString operator+ (const SString& s2) {
+        SString res("");

```

Comment:
Probably, it makes more sense to create this temporary variable using copy c-tor instead of calling append.
return SString{*this}.append(s2.str());